CONTRIBUTING.md

```diff
@@ -85,26 +89,48 @@ To start the emulator and attach an SD card to it, run
 $ANDROID_HOME/tools/emulator -avd bpk-droid-avd -sdcard sd.img &
 ```
 
-After adding new snapshot tests, run
+#### Creating tests
+
+Snapshot tests live in the `app` module. For a new test class extend from `BpkSnapshotTest`.
+Each test function will result in a separate test function - try to add a test for each state of a component.
+Use the `setDimensions` function in the `setup()` function to set the right dimension for the snapshots.
+
+For `View` components you can create the component, set the states and then call `snap` to take the screenshots.
+For `Compose` components use the `composed` function to wrap your component - this will generate the screenshot.
+
+By default screenshot tests run on 4 variants - default, dark mode, RTL and themed (skipped for compose). In some cases you may want to only run a snapshot test on some variants - for example if a component has many different states without layout changes you may want to consider skipping RTL.
+
+You can do this by adding the `assumeVariant(BpkTestVariant.Default, BpkTestVariant.DarkMode)` function with the desired variants. 
+**IMPORTANT**: Make sure this is called _before_ the test activity is run to avoid unnecessary delay. In a view test make it the first line of your test. In a compose test call it before the `composed` function, rather than inside it.
+
+#### Verifying changes
+After adding new snapshot tests or making UI changes, run
 
 ```
  ./scripts/record_screenshot_tests.sh
 ```
 
+This will generate the latest screenshot. Verify the changes & generated screenshots are as expected and commit the changes.
+
 To test changes use
 
 ```
 ./scripts/verify_screenshot_tests.sh
 ```
 
-#### Espresso tests
+If the check fails you either need to fix the issue if a change was unintended or record script above instead to update the screenshots.
+
+### Espresso tests
+If your component contains logic that can't be verified via screenshot tests you can use espresso to test the logic. These tests live in the `Backpack` or `backpack-compose` module, depending on the component.
```

Comment:
tests that live in `Backpack` are for `View` components?

Reply:
i'll clarify that